src/profile/components/basicData/__tests__/BasicData.test.tsx

```diff
@@ -26,36 +35,157 @@ describe('<BasicData /> ', () => {
     );
   };
+  const t = i18n.getFixedT('fi');
+  const editButtonSelector: ElementSelector = { id: 'basic-data-edit-button' };
+  const cancelButtonSelector: ElementSelector = {
+    testId: 'basic-data-cancel-button',
+  };
+
+  beforeEach(() => {
+    responses.length = 0;
+  });
+  afterEach(() => {
+    cleanComponentMocks();
+  });
 
   // verify rendered data
   const verifyValues = async (
     getTextOrInputValue: TestTools['getTextOrInputValue'],
+    source: Partial<BasicDataValue | ProfileData>,
+    targetIsInput = false
+  ) => {
+    const { firstName, nickname, lastName } = source;
+    const getSelector = (name: string): Record<string, string> =>
+      targetIsInput
+        ? { id: `${basicDataType}-${name}` }
+        : { testId: `${basicDataType}-${name}-value` };
+    await expect(getTextOrInputValue(getSelector('firstName'))).resolves.toBe(
+      firstName
+    );
+    await expect(getTextOrInputValue(getSelector('nickname'))).resolves.toBe(
+      nickname
+    );
+    await expect(getTextOrInputValue(getSelector('lastName'))).resolves.toBe(
+      lastName
+    );
+  };
+
+  // set new data to input fields
+  const setValues = async (
+    setInputValue: TestTools['setInputValue'],
     source: Partial<BasicDataValue | ProfileData>
   ) => {
     const { firstName, nickname, lastName } = source;
-    await expect(
-      getTextOrInputValue({ testId: 'basic-data-firstName-value' })
-    ).resolves.toBe(firstName);
-    await expect(
-      getTextOrInputValue({ testId: 'basic-data-nickname-value' })
-    ).resolves.toBe(nickname);
-    await expect(
-      getTextOrInputValue({ testId: 'basic-data-lastName-value' })
-    ).resolves.toBe(lastName);
+    await setInputValue({
+      selector: { id: 'basic-data-firstName' },
+      newValue: firstName as string,
+    });
+    await setInputValue({
+      selector: { id: 'basic-data-nickname' },
+      newValue: nickname as string,
+    });
+    await setInputValue({
+      selector: { id: 'basic-data-lastName' },
+      newValue: lastName as string,
+    });
+  };
+
+  const initTests = async (): Promise<TestTools> => {
+    responses.push({ profileData: initialProfile });
+    const testTools = await renderTestSuite();
+    await testTools.fetch();
+    return Promise.resolve(testTools);
   };
 
-  it("renders user's names", async () => {
-    const responses: MockedResponse[] = [
-      { profileData: getMyProfile().myProfile as ProfileData },
-    ];
+  it("renders user's names - also in edit mode", async () => {
+    await act(async () => {
+      const { getTextOrInputValue, triggerAction } = await initTests();
+      await verifyValues(getTextOrInputValue, initialProfile);
+      // goto edit mode
+      await triggerAction(editButtonSelector);
+      await verifyValues(getTextOrInputValue, initialProfile, true);
+    });
+  });
+  it('sends new data and returns to view mode when saved', async () => {
+    // new values
+    const basicData = {
```

Comment:
This _basicData_ initialisation seems to made in the beginning of each test. I wonder if we could initialise it inside the _beforeEach_?

Reply:
Done.